### DISCUSSION TITLE
HmIPW-WRC6 LEDs controllable

### DISCUSSION THREAD

**Discussion by Trickfilm400** [SukramJ/aiohomematic#1163] [Ideas]
Hi,

We have some [HmIPW-WRC6](https://de.elv.com/homematic-ip-wired-smart-home-wandtaster-hmipw-wrc6-6-fach-154288) in use and I would like to control the LEDs of this device via Homeassistant, but sadly, if I want to change the color, the mode of the LEd is always reset to disabled - A new option in Homeassistant would be needed to add all the LED methods:

<img width="331" alt="image" src="https://github.com/danielperna84/hahomematic/assets/34717401/e55ca3d7-597d-478b-8d66-af7785bb4346">
In this image is shown, which attributes are working with Homeassisant and the "Modus" thing is not working

![image](https://github.com/danielperna84/hahomematic/assets/34717401/464df34b-4c6a-476a-b0da-d8acc71154e1)
Regardless of which setting the mode is, after I change the LED color for example or anything else, the mode is reset to "Beleuchtung aus", which is a disabled or turned off LED

![image](https://github.com/danielperna84/hahomematic/assets/34717401/079dffc3-912e-478f-b4e9-8076b0fd1f83)
In Homeassistant it looks like this

Would be great if there would be a solution to control the mode via homeassistant

==================================

Ich würde gerne die LEDs von diesen o. g. Wandtastern steuern, aber HA überschreibt den Modus jedesmal auf "Beleuchtung aus", wodurch die Steuerung überhaupt nicht funktioniert
Gbit es dazu irgendwie eine Lösung?

Grüße

**Comment by SukramJ** [comment]
Da ich selber keinen HmIPW-WRC6 basiert die aktuelle Implementierung nur auf der Spezifikation.
Die aktuelle Implementierung unterstützt keine Effekte/Modi.

> Gbit es dazu irgendwie eine Lösung?

Klar, man muss die Implementierung anpassen.

Da mir aus deinem Text nicht klar wird, was derzeit schon funktioniert, hier ein Punkte, die ich bei der aktuellen Implementierung erwarten würde.

- Es gibt für jeden der 6 Kanäle eine `light` Entität.
- Für jeden Kanal kann die Farbe mit der `light` Entität ausgewählt werden, wobei der Farbwert zur unterstützen Farbe springt.
- Für jeden Kanal kann die Helligkeit eingestellt werden

Ab hier geht es nur um Einstellungen in der CCU:

Da mir die Funktionsweise der Modi nicht klar ist, bräuchte ich dazu ein möglichst klare Beschreibung.
Was soll `Beleuchtung an` bzw. `Beleuchtung aus`? Reicht es nicht die Helligkeit zu erhöhen und eine Farbe setzen, damit das Licht angeht, bzw. Helligkeit auf 0, damit das Licht wieder ausgeht?
Was ändert sich an den Modi, wenn die Helligkeit auf 0 bzw  > 0 gestellt wird.

Also die grobe Frage ist wie spielen Änderungen von Helligkeit und Farbe mit den Modi zusammen, bzw. wie beeinflussen sich diese gegenseitig?

**Reply by Trickfilm400** [reply]
> Da mir aus deinem Text nicht klar wird, was derzeit schon funktioniert, hier ein Punkte, die ich bei der aktuellen Implementierung erwarten würde.
> 
> Es gibt für jeden der 6 Kanäle eine light Entität.
> Für jeden Kanal kann die Farbe mit der light Entität ausgewählt werden, wobei der Farbwert zur unterstützen Farbe springt.
> Für jeden Kanal kann die Helligkeit eingestellt werden

Ja genau, das stimmt soweit auch 👍🏻  (-> Ist aktuell so in HA und man kann diese Werte auch setzen, diese werden auch in der CCU übernommen, also wenn ich die Farbe auf 100% Violett stelle ist in der CCU der Wert ebenfalls Violett auf 100%, quasi so, wie im ersten Post der erste und letzte Screenshot zeigen)

> Da mir die Funktionsweise der Modi nicht klar ist, bräuchte ich dazu ein möglichst klare Beschreibung.

Die Modi stellen den Modus der LED ein, wie diese eingeschaltet werden soll - steht der Modus auf `(...) blinken` und die LED wird eingeschaltet, blinkt die LED in der eingestellten Farbe und Helligkeit; so ist das auch mit den Modi `blitzen` und `pulsieren`.
Der Modus `Dauerhaft ein` konfiguriert die LED einfach auf `ein`, die LED leuchtet dann dauerhaft auf der eingestellten Farbe und Helligkeit, sofern die LED eingeschaltet ist (mit dem Ein/Aus Button in der CCU rechts oben (s. 1. Screenshot).
Der Modus `Beleuchtung aus` schaltet die LED aus, egal was bei allen anderen Einstellungen konfiguriert wurde: Die LED ist auf Violett und 100% Helligkeit -> Die LED bleibt aus, da der Modus auf `Beleuchtung aus` steht.

> Was soll Beleuchtung an bzw. Beleuchtung aus? Reicht es nicht die Helligkeit zu erhöhen und eine Farbe setzen, damit das Licht angeht, bzw. Helligkeit auf 0, damit das Licht wieder ausgeht?

Nein - wenn `Beleuchtung an` gewählt wurde, würde die LED normal funktionieren, wenn `Beleuchtung aus` bleibt die LED aus, egal was man auswählt.

> Was ändert sich an den Modi, wenn die Helligkeit auf 0 bzw > 0 gestellt wird.

Ich weiß nicht, ob ich verstehe, was mit der Frage gemeint ist; der Modus ist unabhängig von den anderen LED Einstellungen, also man kann den Modus beliebig wählen und die LED Einstellungen (Farbe, Helligkeit) werden auf den Modus angepasst (Violett, 50% -> Langsames blinken)
Wenn man in der CCU die Helligkeit ändert, ändert sich der Modi nicht.

> Also die grobe Frage ist wie spielen Änderungen von Helligkeit und Farbe mit den Modi zusammen, bzw. wie beeinflussen sich diese gegenseitig?

In der CCU funktioniert alles problemlos. Man kann alle Einstellungen treffen und nichts überschreibt etwas. Der Modus beeinflusst die Farbe und Helligkeit quasi so, dass der Modus vorgibt, wie die LED sich verhalten soll, ob die LED blinken soll oder nicht - in der konfigurierten Farbe und Helligkeit

Aber das Problem mit Homeassistant: Wenn ich in HA die Farbe oder Helligkeit ändere, wird der Modus (von HA wahrscheinlich im Hintergrund) immer auf `Beleuchtung aus` gesetzt (das müsste der Standard-Wert für den Modus sein, da die automatisch diesen Modus ausgewählt hatten nach der Ersteinrichtung). So lässt sich natürlich die HA Steuerung nicht nutzen, weil egal was man einstellt, die LED bleibt aus aufgrund des zurücksetzen des Modus. (Wenn ich in der CCU den Modus auf irgendetwas setze, z. B. `Beleuchtung ein` und ändere die Farbe in HA wird der Modus erneut auf `Beleuchtung aus` gesetzt)

Ich hoffe das klärt einige Fragen, ansonsten gerne nochmal fragen :)

**Reply by SukramJ** [reply]
Danke.
Jetzt muss mal mal schauen, wie das zu HA Semantik passt.
Also ein erster Schritt in HA wäre:
- Wenn das Gerät ausgeschaltet wird (Helligkeit ist 0), dann wird der Modus auf `Beleuchtung aus` gesetzt.
- Wenn das Gerät eingeschaltet wird (Helligkeit > 0), dann wird der Modus auf` Beleuchtung ein` gesetzt.
- Bei den Modi würde ich `Beleuchtung aus` und `Beleuchtung ein` im Dropdown (Neu) ausblenden, da die Steuerung durch andere Steuerelemente erfolgt.
- Wenn zwischen den restlichen Modi gewählt wird, dann werden diese angewendet, haben aber keinen Einfluss auf an/aus.

Gibt es in der CCU die Farbwerte `OLD_VALUE` und `DO_NOT_CARE` ?

**Reply by FlowFrank** [reply]
> > Da mir aus deinem Text nicht klar wird, was derzeit schon funktioniert, hier ein Punkte, die ich bei der aktuellen Implementierung erwarten würde.
> 
> > 
> 
> > Es gibt für jeden der 6 Kanäle eine light Entität.
> 
> > Für jeden Kanal kann die Farbe mit der light Entität ausgewählt werden, wobei der Farbwert zur unterstützen Farbe springt.
> 
> > Für jeden Kanal kann die Helligkeit eingestellt werden
> 
> 
> 
> Ja genau, das stimmt soweit auch 👍🏻  (-> Ist aktuell so in HA und man kann diese Werte auch setzen, diese werden auch in der CCU übernommen, also wenn ich die Farbe auf 100% Violett stelle ist in der CCU der Wert ebenfalls Violett auf 100%, quasi so, wie im ersten Post der erste und letzte Screenshot zeigen)
> 
> 
> 
> > Da mir die Funktionsweise der Modi nicht klar ist, bräuchte ich dazu ein möglichst klare Beschreibung.
> 
> 
> 
> Die Modi stellen den Modus der LED ein, wie diese eingeschaltet werden soll - steht der Modus auf `(...) blinken` und die LED wird eingeschaltet, blinkt die LED in der eingestellten Farbe und Helligkeit; so ist das auch mit den Modi `blitzen` und `pulsieren`.
> 
> Der Modus `Dauerhaft ein` konfiguriert die LED einfach auf `ein`, die LED leuchtet dann dauerhaft auf der eingestellten Farbe und Helligkeit, sofern die LED eingeschaltet ist (mit dem Ein/Aus Button in der CCU rechts oben (s. 1. Screenshot).
> 
> Der Modus `Beleuchtung aus` schaltet die LED aus, egal was bei allen anderen Einstellungen konfiguriert wurde: Die LED ist auf Violett und 100% Helligkeit -> Die LED bleibt aus, da der Modus auf `Beleuchtung aus` steht.
> 
> 
> 
> > Was soll Beleuchtung an bzw. Beleuchtung aus? Reicht es nicht die Helligkeit zu erhöhen und eine Farbe setzen, damit das Licht angeht, bzw. Helligkeit auf 0, damit das Licht wieder ausgeht?
> 
> 
> 
> Nein - wenn `Beleuchtung an` gewählt wurde, würde die LED normal funktionieren, wenn `Beleuchtung aus` bleibt die LED aus, egal was man auswählt.
> 
> 
> 
> > Was ändert sich an den Modi, wenn die Helligkeit auf 0 bzw > 0 gestellt wird.
> 
> 
> 
> Ich weiß nicht, ob ich verstehe, was mit der Frage gemeint ist; der Modus ist unabhängig von den anderen LED Einstellungen, also man kann den Modus beliebig wählen und die LED Einstellungen (Farbe, Helligkeit) werden auf den Modus angepasst (Violett, 50% -> Langsames blinken)
> 
> Wenn man in der CCU die Helligkeit ändert, ändert sich der Modi nicht.
> 
> 
> 
> > Also die grobe Frage ist wie spielen Änderungen von Helligkeit und Farbe mit den Modi zusammen, bzw. wie beeinflussen sich diese gegenseitig?
> 
> 
> 
> In der CCU funktioniert alles problemlos. Man kann alle Einstellungen treffen und nichts überschreibt etwas. Der Modus beeinflusst die Farbe und Helligkeit quasi so, dass der Modus vorgibt, wie die LED sich verhalten soll, ob die LED blinken soll oder nicht - in der konfigurierten Farbe und Helligkeit
> 
> 
> 
> Aber das Problem mit Homeassistant: Wenn ich in HA die Farbe oder Helligkeit ändere, wird der Modus (von HA wahrscheinlich im Hintergrund) immer auf `Beleuchtung aus` gesetzt (das müsste der Standard-Wert für den Modus sein, da die automatisch diesen Modus ausgewählt hatten nach der Ersteinrichtung). So lässt sich natürlich die HA Steuerung nicht nutzen, weil egal was man einstellt, die LED bleibt aus aufgrund des zurücksetzen des Modus. (Wenn ich in der CCU den Modus auf irgendetwas setze, z. B. `Beleuchtung ein` und ändere die Farbe in HA wird der Modus erneut auf `Beleuchtung aus` gesetzt)
> 
> 
> 
> Ich hoffe das klärt einige Fragen, ansonsten gerne nochmal fragen :)

Kann ich im Großen und Ganzen so bestätigen. Habe selber mehrere der Taster im Einsatz und dieses Thema stört mich auch auch schon seit längerem. Ich hatte ehrlich gesagt aber noch nicht die Musse wie mein Vorredner, das Thema hier aufzumachen. Allein dafür schon mal vielen Dank!

**Reply by Trickfilm400** [reply]
@FlowFrank Das ist schön zu hören :)

> Gibt es in der CCU die Farbwerte `OLD_VALUE` und `DO_NOT_CARE` ?

@SukramJ 
Wo genau müsste ich nach diesen Werten suchen? mit den Browser Devtools bin ich bisher noch nicht fündig geworden, dort wird der Farbwert im Request als integer mitgegeben, aber ich weiß nicht ob das die richtige Stelle zum suchen ist


Ansonsten hört sich der Plan mit Homeassistant und dem Dropdown zuversichtlich an 👍🏻

**Reply by SukramJ** [reply]
> Wo genau müsste ich nach diesen Werten suchen? 

Übersetzt im Farbe Dropdown in der CCU UI. Wenn der Wert nicht da ist wir dieser nur intern verwendet.

**Reply by Trickfilm400** [reply]
> Übersetzt im Farbe Dropdown in der CCU UI. Wenn der Wert nicht da ist wir dieser nur intern verwendet.

Hier ein Screenshot der Farb-Auswahl:

![image](https://github.com/danielperna84/hahomematic/assets/34717401/6c2b971d-8c1e-4236-a38f-4be0e149b028)

Und der dazugehörige HTML Auszug:

![image](https://github.com/danielperna84/hahomematic/assets/34717401/00ca0a5e-cf6f-45bd-b96e-d5c0a5ac6399)

Und im `datapointconfigurator.htm` Request Response kommt folgendes:
```html
(...)

<select id='bslColorSelector2811' onchange='setDpState(2813, this.value);' style='text-align:center;'>
<option value=0>${lblOff}</option>
<option value=1>${optionColorBLUE}</option>
<option value=2>${optionColorGREEN}</option>
<option value=3>${optionColorTURQUOISE}</option>
<option value=4>${optionColorRED}</option>
<option value=5>${optionColorPURPLE}</option>
<option value=6>${optionColorYELLOW}</option>
<option value=7>${optionColorWHITE}</option>
</select>

(...)

<select id='behaviourSelector2811' onchange='setDpState(2814, this.value);' style='text-align:center;'>
<option value=0>${optionColorOFF}</option>
<option value=1>${optionColorON}</option>
<option value=2>${optionBlinkSlow}</option>
<option value=3>${optionBlinkMiddle}</option>
<option value=4>${optionBlinkFast}</option>
<option value=5>${optionBlinkFlashSlow}</option>
<option value=6>${optionBlinkFlashMiddle}</option>
<option value=7>${optionBlinkFlashFast}</option>
<option value=8>${optionBlinkBillowSlow}</option>
<option value=9>${optionBlinkBillowMiddle}</option>
<option value=10>${optionBlinkBillowFast}</option>

(...)
```

Hilft das oder ist das was anderes?

**Reply by SukramJ** [reply]
Das hilft.

**Reply by Trickfilm400** [reply]
Als reine Information: Da ich von Python nicht wirklich eine Ahnung habe kann ich bei der Implementierung nicht wirklich helfen, aber ich kann beim Testen helfen oder mit zusätzlichen Informationen - Würde mich daher freuen, wenn sich jemand der Implementierung annimmt
(Ich komme aus der Nodejs-Richtung und meine Python Kenntnisse sind sehr grundlegend)

**Reply by SukramJ** [reply]
Implementierung mach ich schon.

**Reply by SukramJ** [reply]
Teste bitte mal Version 1.40.2b3.
Ggf. musst Du dazu in HACS an der Integration  unter `erneut herunterladen` die beta Versionen aktivieren.

**Reply by Trickfilm400** [reply]
Hi,
Update ist installiert, LEDs sind steuerbar, ist sehr schön 
Einige Sachen sind mir aufgefallen, wo ich nicht weiß, ob das so soll oder nicht
1. Wenn man die Lampe ausschaltet und neu einschaltet (über den Haupt-Ein_Aus-Schalter, der mit dem Powersymbol) - dann bleiben die Einstellungen nicht erhalten und die Lampe ist erneut Weiß mit 100% Helligleit
Ich habe mal geschaut, das liegt daran, dass beim Ausschalten die Parameter wie Farbe auf "Aus" und die Helligkeit auf 0% gestellt werden in der CCU, obwohl man beim ausschalten nur den "aus" Knopf betätigen können würde. Das wäre eine schöne Verbesserung, da die LED Einstellungen über ein ein/ausschalten hinweg die gleichen blieben und man nicht neu doe Farbe und Helligkeit einstellen müsste (sofern das keine Limitierung von HA ist, dann ist das ganze natürlich egal)
2. Werden die Effekte in HA noch übersetzt? Mich stört das nicht, wäre wahrscheinlich nur ganz sinnvoll, wenn jemand mit den englischen Begriffen in caps lock nichts anfangen kann
3. Ähnlich zu Punkt 1: wenn man eine Effekt eingestellt hat und die Farbe wechselt oder die Helligkeit anpasst wird der Modus zurück auf "dauerhaft ein" gesetzt, was nicht ganz so praktisch wäre

Das sollten die Punkte gewesen sein, die mir aufgefallen waren, ich hoffe ich habe nichts vergessen
Danke schonmal für die Implementierung

**Reply by SukramJ** [reply]
Danke für die Rückmeldung.
zu 1. welcher Modus wird denn beim Einschalten in der CCU gesetzt?
zu 2. nicht von mir, aber kannst du ja gerne hier anhängen.

**Reply by Trickfilm400** [reply]
Beim einschalten wird die Farbe weiß auf 100% im Modus "dauerhaft ein" eingestellt in der ccu

Beim ausschalten wird die Farbe auf "aus" mit 0% Helligkeit und dem Modus "Beleuchtung aus" eingestellt in der ccu

(Am Handy funktioniert der Screenshot upload warum auch immer nicht, sonst wäre das einfacher gewesen)

**Reply by SukramJ** [reply]
Dann ist das Ergebnis bei Verwendung der CCU identisch mit HA aber anders als bei der manuellen Betätigung?

**Reply by Trickfilm400** [reply]
![Screenshot_20230824_162608_Chrome.jpg](https://github.com/danielperna84/hahomematic/assets/34717401/d9c1d167-03eb-44c2-bdbc-1b3589d3d27c)

Also wenn man da auf "aus" drückt wird die Helligkeit auf 0% gestellt und wenn man auf ein drückt ist die Helligkeit bei 100% - sorry dafür, das hatte ich anders auf dem Schirm - das passt ja dann mit HA

Allerdings sieht man, dass der Modus und die Farbe gespeichert bleibt, wenn man die LED aus/einschaltet

Das ist mit HA ja aktuell nicht der Fall, da wird es immer auf dauerhaft ein mut der Farbe weiß gesetzt 

Was meinst du ansonsten mit der "manuellen Betätigung"? (Es gibt ja nur die ccu und HA als Steuerung, bin verwirrt)

**Reply by SukramJ** [reply]
> Was meinst du ansonsten mit der "manuellen Betätigung"? (Es gibt ja nur die ccu und HA als Steuerung, bin verwirrt)

Mein Fehler. Einen Geräteschalter gibt es ja nicht.

**Reply by SukramJ** [reply]
> Also wenn man da auf "aus" drückt wird die Helligkeit auf 0% gestellt und wenn man auf ein drückt ist die Helligkeit bei 100% - sorry dafür, das hatte ich anders auf dem Schirm - das passt ja dann mit HA

Ich es noch mal für mich zusammen zu fassen:

Beim Einschalten:
- Die Helligkeit wird auf 100% gesetzt und der Modus wird auf `Ein` gesetzt, falls `Aus` oder nicht definiert und die Farbe auf `Weiss`, falls `Schwarz` oder nicht definiert.
Einschalten über Helligkeit:
- Die Helligkeit wird auf den gewünschten Wert gesetzt und der Modus wird auf `Ein` gesetzt, falls `Aus` oder nicht definiert und die Farbe auf `Weiss`, falls `Schwarz` oder nicht definiert.

Beim Ausschalten:
- Die Helligkeit wird auf 0% gesetzt. Modus und Farbe bleiben unverändert

**Reply by Trickfilm400** [reply]
Beim Einschalten:
Wenn man in der CCU auf "ein" drückt wird die Helligkeit auf 100% gesetzt - mehr nicht
Wenn die Farbe auf "aus" steht, muss man diese ändern, damit eine Farbe erscheint
Wenn der Modus auf "Beleuchtung aus" steht, muss man diesen zusätzlich ändern, damit die LED final angeht

Bei der CCU muss man die Farbe und den Modus manuell (zusätzlich) zur Helligkeit anpassen, um die LED einzuschalten
Wenn die Farbe z. B. Auf "aus" steht, kann der Modus irgendwas sein oder die helligkeit irgendwas sein, die LED bleibt aus und geht nicht an, bis man die Farbe umstellt


> Die Helligkeit wird auf den gewünschten Wert gesetzt und der Modus wird auf Ein gesetzt, falls Aus oder nicht definiert und die Farbe auf Weiss, falls Schwarz oder nicht definiert.

Das ist aktuell in HA so, was ja aber nicht der CCU Steuerung entspricht, da es dort manuell geändert werden muss; das Ziel ist ja aber, das die Farbe nur gesetzt wird, wenn sie vorher "aus" war, damit die LED angeht, ansonsten sollte ja die vorherige Farbe verwendet werden


Beim ausschalten: das stimmt

**Reply by SukramJ** [reply]
> Wenn man in der CCU auf "ein" drückt wird die Helligkeit auf 100% gesetzt - mehr nicht
Wenn die Farbe auf "aus" steht, muss man diese ändern, damit eine Farbe erscheint
Wenn der Modus auf "Beleuchtung aus" steht, muss man diesen zusätzlich ändern, damit die LED final angeht

Das ist ja grauenvoll.
Es ist nicht das Ziel die CCU abzubilden.
Kannst Du dich mit meinen Vorschlägen zu Einschalten anfreunden?

**Reply by Trickfilm400** [reply]
Ja, das sollte auf jeden Fall passen, mir fällt kein Problem auf

(Für den Fall, dass man automationen verwendet, hat man das Problem ja sowieso nicht, geht ja nur drum wenn man die Lampen manuell steuert)

**Reply by FlowFrank** [reply]
> Ja, das sollte auf jeden Fall passen, mir fällt kein Problem auf
> 
> (Für den Fall, dass man automationen verwendet, hat man das Problem ja sowieso nicht, geht ja nur drum wenn man die Lampen manuell steuert)

Falls ich heute Abend noch ein bisschen Zeit finde, dann probiere ich das auch gerne einmal aus. Wie gesagt, ich möchte schon lange die LEDs in allen Räumen auch zur Anzeige von irgendwelchen Status nutzen, aber habe nie die Musse gehabt mich darum zu kümmern. 
Ich finde das so cool, dass du @SukramJ das so schnell mal eben umgesetzt hast! Hammer!

Für eine Sache nutze ich die LEDs tatsächlich. In der Küche lasse ich durch die sechs LEDs anzeigen, wie viel Watt unser Balkonkraftwerk gerade liefert. Das mache ich aber tatsächlich über die Homematic IP Schalt-Mess-Steckdose direkt über eine Automation in der CCU.

**Reply by SukramJ** [reply]
Die Version 1.40.2b4 enthält die letzten Änderungen.
Ich bitte um Feedback.

**Reply by Trickfilm400** [reply]
Jede Änderung der Helligkeit stellt den Modus um: wenn der Modus auf ON steht und ich ändere die Helligkeit wird die Helligkeit zwar gesetzt, der Modus aber auf OFF. ändert man die Helligkeit erneut, stellt sich der Modus wieder auf ON, und so im Wechsel. Wenn der Modus am Anfang auf einem anderen beliebigen Wert stand wird er auf OFF gesetzt bei einer Änderung der Helligkeit 

Dieses Problem existiert auch wenn man den Power Button der LED entity nutzt: schaltet man die LED ein wird der Modus auf ON gesetzt, schaltet man die LED aus, ist die Effekt Schaltfläche ohne Wert, was ja passt, (Standard HA Verhalten); schaltet ich die LED jetzt ein, stellt sich der Modus auf OFF, die LED bleibt aus

Okay das Problem ist auch beim Ändern der Farbe, jedes mal wenn man die Farbe wechselt ändert sich der Modus von OFF  zu ON oder umgekehrt 

Positiv ist: die Farbe wird gesichert zwischen dem ein/ausschalten,  das funktioniert gut 

Der Helligkeitswert wird ja nicht gesichert zwischen ein/ausschalten der LED, korrekt? (So hatte ich das verstanden) 

Sollte der Modus gespeichert werden zwischen ein/ausschalten? Also wenn man blinken ausgewählt hat, dass dann nach aus und wieder einschalten der LED der Modus wieder blinken ist?

> der Modus wird auf Ein gesetzt, falls Aus oder nicht definiert

Das hatte ich so verstanden, dass der Modus nur gesetzt wird, wenn er aus oder was anderes war, aber wenn der Modus vorher z. B. Auf blinken stand, dass der Modus dann nicht neu gesetzt wird

Aber gut möglich,  dass das mit dem Bug wie oben beschrieben steht, zusammenhängt

**Reply by SukramJ** [reply]
> Jede Änderung der Helligkeit stellt den Modus um: wenn der Modus auf ON steht und ich ändere die Helligkeit wird die Helligkeit zwar gesetzt, der Modus aber auf OFF. ändert man die Helligkeit erneut, stellt sich der Modus wieder auf ON, und so im Wechsel. Wenn der Modus am Anfang auf einem anderen beliebigen Wert stand wird er auf OFF gesetzt bei einer Änderung der Helligkeit

Das sind Änderungen, die nicht durch den HA Code ausgelöst werden.
Wenn Du dir den dazugehörigen [Test](https://github.com/SukramJ/hahomematic/blob/dev1/tests/test_light.py#L505) anschaust sollte das geplante Vorhaben dort zu sehen sein.
Der Modus wird nur beim Einschalten auf "ON" gesetzt, wenn der Modus auf `OFF`, `DO_NOT_CARE` oder `OLD_VALUE`  steht.
Ansonsten wird der Modus nur durch eine Änderung des `effect` in HA geändert.

Ich bin mir noch nicht im Klaren wie die Modi `DO_NOT_CARE` oder `OLD_VALUE` (die Werte gibt es auch bei den Farben) verwendet werden müssen. Vorstellbar wäre das beim Ausschalten auf `OLD_VALUE` gesetzt werden muss, aber das werden wir so nicht rauskriegen.

Am Besten wäre, wenn Du dir den [Homematic Manager](https://github.com/hobbyquaker/homematic-manager/releases/tag/2.7.1) installieren könntest, dann kann man direkt auf die API werte schauen, und sieht dann auch wie mit `DO_NOT_CARE` oder `OLD_VALUE` umgegangen wird.

**Reply by SukramJ** [reply]
> schaltet man die LED aus, ist die Effekt Schaltfläche ohne Wert, was ja passt

Das spricht dafür das DO_NOT_CARE oder OLD_VALUE verwendet wird.

Testweise (alternativ zum Homematic Manager) könntest Du mal [unignore](https://github.com/danielperna84/custom_homematic#unignore-device-parameters) nutzen.

Und die Parameter COLOR und COLOR_BEHAVIOUR eintragen und HA neustarten.

Dadurch werden zusätzliche Entiäten angelegt, die die jeweiligen Rohwerte anzeigen.

**Reply by Trickfilm400** [reply]
![image](https://github.com/danielperna84/hahomematic/assets/34717401/b6d5f99b-d209-4c5e-bb5f-f7714f0900c0)


![image](https://github.com/danielperna84/hahomematic/assets/34717401/8f3f3696-45dd-4121-903b-cd1a51ce74c1)


Dieser Zustand ist, als ich die LED in HA mit dem Power-Button da ausgeschaltet habe

Man sieht, dass die Helligkeit auf 0% ist -> aus
und die Farbe ist gespeichert


----

![image](https://github.com/danielperna84/hahomematic/assets/34717401/ad9d834c-d093-440b-a001-08ff9d16870c)


Schalte ich die LED jetzt an, wird der Modus auf aus gesetzt

![image](https://github.com/danielperna84/hahomematic/assets/34717401/cf6bd68c-1bf5-4a0a-acae-8a4cc323b090)

----

Laut Homematic Manager gibt es `DO_NOT_CARE` und `OLD_VALUE` 
![image](https://github.com/danielperna84/hahomematic/assets/34717401/899a36dc-318f-45a1-9295-323b84729f08)
![image](https://github.com/danielperna84/hahomematic/assets/34717401/a6d2a277-74c9-4711-adc8-7548c9721c44)



----

> > schaltet man die LED aus, ist die Effekt Schaltfläche ohne Wert, was ja passt

> Das spricht dafür das DO_NOT_CARE oder OLD_VALUE verwendet wird.

Bevor ich das anders verstanden habe, hier ein screenshot wie das aussieht, was ich meinte (nicht,  dass das vo Ha einfach nur so angezeigt wird)
![image](https://github.com/danielperna84/hahomematic/assets/34717401/f6078502-2e7e-462f-b5fc-3b67c5651282)


----

> Der Modus wird nur beim Einschalten auf "ON" gesetzt, wenn der Modus auf OFF, DO_NOT_CARE oder OLD_VALUE steht.
Ansonsten wird der Modus nur durch eine Änderung des effect in HA geändert.

Das ist sehr seltsam, weil der Wert steht z. B. lauf CCU usw. auf einem blink-Modi und HA setzt den Modus beim einschalten trotzdem neu

**Reply by SukramJ** [reply]
> Schalte ich die LED jetzt an, wird der Modus auf aus gesetzt

Mir ist nicht immer ganz klar wo du was schaltest. In HA oder in der CCU.

Wenn Du über HA oder der CCU die Helligkeit, die Farbe oder den Modus änderst, kannst Du dann jemals im HM-Manager sehen das bei COLOR und/oder COLOR_BEHAVIOUR die Werte DO_NOT_CARE oder OLD_VALUE auftauchen?

Ich werde bei der nächsten Version den aktuellen Modus immer mitgeben.

**Reply by Trickfilm400** [reply]
> Mir ist nicht immer ganz klar wo du was schaltest. In HA oder in der CCU.

Ah sorry, ich habe immer über HA gesteuert

> Wenn Du über HA oder der CCU die Helligkeit, die Farbe oder den Modus änderst, kannst Du dann jemals im HM-Manager sehen das bei COLOR und/oder COLOR_BEHAVIOUR die Werte DO_NOT_CARE oder OLD_VALUE auftauchen?

Ich habe in HA die LED eingeschaltet
Folgendes passiert im HM-Manager wenn ich das VALUES-Popup offen habe;
![image](https://github.com/danielperna84/hahomematic/assets/34717401/7ec6f2de-982e-4a70-b448-7dcf117367d5)
Die Werte verschwinden alle
Wenn ich das Popup neu öffne, sind die Werte wieder da
![image](https://github.com/danielperna84/hahomematic/assets/34717401/8868c4c7-462e-42a1-bc25-80c4dbc094f0)

Ich weiß nicht ob das ein Problem von dem Programm ist, ich gehe mal davon aus, aber das weiß ich nicht, wie das gedacht ist (okay jetzt passiert nichts mehr wenn ich in HA was drücke, ich muss das popup immer neu aufmachen, also ich glaube das hat hiermit nichts zu tun)

Dann habe ich in HA die LED ausgeschaltet und folgendes ist in HM-Manager zu sehen:
![image](https://github.com/danielperna84/hahomematic/assets/34717401/94e7b95d-76c2-436d-a497-7203ccbe71a6)

Man erkennt am `LEVEL`, dass die LED aus ist

Schalte ich die LED in HA wieder an, ist `LEVEL` bei 100, aber der Modus auf OFF:
![image](https://github.com/danielperna84/hahomematic/assets/34717401/06d0f663-1b27-428a-a9fb-391afc4cba70)


> kannst Du dann jemals im HM-Manager sehen das bei COLOR und/oder COLOR_BEHAVIOUR die Werte DO_NOT_CARE oder OLD_VALUE auftauchen?

Ich sehe diese Werte nicht auftauchen (sofern ich das Programm richtig verwende)  / Falls sie in diesem Popup auftauchen sollten (Ich bin im HM-Manager auf Geräte gegangen, habe dort vom Schalter den Farbkanal gesucht und per Rechtsklick auf `VALUES Paramset` gedrückt

**Reply by SukramJ** [reply]
Mich wundert das COLOR_BEHAVIOUR in deinem ersten Bild leer ist.

Macht es wirklich Sinn das als Effekt der Modus OFF angezeigt wird? Ich würde das wieder ausblenden.
Bei der nächsten Version wird, wie gesagt, COLOR_BEHAVIOUR immer mitgeschickt.

**Reply by Trickfilm400** [reply]
> Mich wundert das COLOR_BEHAVIOUR in deinem ersten Bild leer ist.

Ja, wie gesagt, das war bei den ersten malen so, sobald ein Update durch HA gekommen ist wurden die Werte geleert, bis man das Popup neu geöffnet hatte, aber ich weiß nicht woran das liegt

> Macht es wirklich Sinn das als Effekt der Modus OFF angezeigt wird? Ich würde das wieder ausblenden.

Ja, das kann man ausblenden, ich sehe da keinen Vorteil hinter

> Bei der nächsten Version wird, wie gesagt, COLOR_BEHAVIOUR immer mitgeschickt.

Das sollte das Problem ja lösen, oder? Die Frage ist dann, bleibt der Vorherige Modus dann erhalten, wenn man die LED aus und wieder einschaltet?

**Reply by SukramJ** [reply]
Bitte mal die 1.40.2b5 testen.

**Reply by Trickfilm400** [reply]
Sieht sehr gut aus! Ich glaube das passt jetzt, die Effekte bleiben so, wie sie vorher gesetzt waren, die Farbe wird auch übernommen

Ich finde jetzt keine Fehler aktuell

**Reply by SukramJ** [reply]
Sehr schön, vielen Dank fürs Testen und die guten Rückmeldungen.

**Reply by Trickfilm400** [reply]
Gerne, ich danke auch für sie Implementierung und Umsetzung!

----

Eine kleine offtopic frage hätte ich noch, was die taster betrifft von den Geräten 

![Screenshot_20230825_134206_Home Assistant.jpg](https://github.com/danielperna84/hahomematic/assets/34717401/db7622f5-30de-4a80-8317-7257d3870eec)

Es werden bei Ereignissen diese taster angezeigt, aber wenn ich einen Knopf drücke wird das in HA nicht angezeigt

Soll das so?

Ich kann dafür eine neue Diskussion aufmachen, wenn das eine größere Sache wäre
Ansonsten würde ich diese Diskussion schließen, da es ja fertig ist.

**Reply by SukramJ** [reply]
Infos findest Du [hier](https://github.com/danielperna84/custom_homematic#events-for-homematicip-devices).

**Reply by Trickfilm400** [reply]
Ach ja, genau, da war ja was, für die Fernbedienung hatte ich das sogar schonmal gemacht 

Hatte irgendwie den falschen Abschnitt in der documentation offen

Vielen Dank für den Hinweis!
